osrf/vrx-docker
# New yaml for generating practice tasks 1-3

This PR adds yaml files for the stationkeeping, wayfinding and perception tasks, updated to support VRX 2022. 

## Notes:
* The 3 yaml files are intended to generate practice worlds that match the ones already stored in the VRX repository under the `practice_2022` directory.
* I will create a separate PR for tasks 4-6.
* The process of generating worlds relies on xacro files in the VRX repository. These currently still have the projectile bowl in them, so the generated worlds will also. Once [VRX PR 380](https://github.com/osrf/vrx/pull/380) is merged the generated worlds won't have the bowl anymore.
* I moved the yaml for VRX 2019 to a separate subfolder to avoid confusion.
* I deleted the practice worlds and xacros for 2019 that were generated from the 2019 yaml and cached in this repository. They are also cached in the VRX repository, so not needed here.

## How to test:
These instructions assume you have a workspace at `~/vrx_ws` with both vrx and vrx-docker installed. Then run through the following for each of the three tasks:
* First, switch to the root of the `vrx-docker` repository.  
  ```
  cd ~/vrx_ws/src/vrx-docker
  ```
* Set some variables for convenience:
  ```
  TASK=stationkeeping
  TRIAL=0
  ```
* Generate the trial worlds for the task
  ```
  ./prepare_task_trials.bash $TASK
  ```
* Verify that 3 worlds are generated without errors (except the expected `REQUIRED process [world_gen-2] has died!` message).
* Now for each trial, 0-2
    * Test that the generated world runs:
      ```
      roslaunch vrx_gazebo vrx.launch world:=`pwd`/generated/task_generated/${TASK}/worlds/${TASK}${TRIAL}.world verbose:=true wamv_locked:=true
      ```
    * Check that the generated world matches the corresponding world in `vrx/vrx_gazebo/worlds/2022_practice` in all important details. The way I do this is by looking at the diff with vimdiff. For example:
      ```
      cd ~/vrx_ws/src/
      vimdiff vrx-docker/generated/task_generated/${TASK}/worlds/${TASK}${TRIAL}.world 
      vrx/vrx_gazebo/worlds/2022_practice/${TASK}${TRIAL}.world
      ```
